modules/parser/src/parsers/parse-stream-data-message.js

```diff
@@ -79,13 +91,7 @@ export function parseStreamDataMessage(message, onResult, onError, opts) {
   }
 
   try {
-    let data;
-    if (typeof message === 'string') {
-      data = JSON.parse(message);
```

Comment:
What is the motivation for this change - Is there a user case where message is string but not JSON string?

Reply:
Because when we send the data in the shape of JSON or Uint8Array to parse function, and when web worker is used, the data is mutated.

The goal is to save a snapshot of data that can't be mutated and with minimal perf impact.